charmcraft/commands/store/__init__.py

```diff
@@ -30,6 +30,17 @@
 logger = logging.getLogger('charmcraft.commands.store')
 
 
+def get_name_from_metadata():
+    """Return the name (if present) from metadata file (if there and readable and sane)."""
```

Comment:
this is ok, but would it be better to actually just load all of metadata, and keep it around, so that things wouldn't have to parse it multiple times to get a single piece from it?
I suppose that could come in a later refactoring.

Reply:
We only get the name from it, so far. And yes, we get from two different places, but as those are different commands, they are not run "at the same time". 

IOW, we only parse the metadata once per charmcraft process execution.